packages/inertia-vue/src/form.js

```diff
@@ -0,0 +1,99 @@
+import Vue from 'vue'
+import { Inertia } from '@inertiajs/inertia'
+
+export default data => {
+  const form = {
+    data: { ... data },
+    defaults: { ... data },
+    errors: {},
+    processing: false,
+    submit(method, url, options) {
+      url = url || Inertia.page.url
+      options = options || {}
+
+      Inertia[method](url, form.data, {
+        ... options,
+        onStart: visit => {
+          form.processing = true
+
+          if (options.onStart) {
+            return options.onStart(visit)
+          }
+        },
+        onFinish: () => {
+          form.processing = false
+
+          if (options.onFinish) {
+            return options.onFinish()
+          }
+        },
+        onError: errors => {
+          Vue.set(form, 'errors', errors)
+
+          if (options.onError) {
+            return options.onError(errors)
+          }
+        },
+        onSuccess: page => {
+          Vue.set(form, 'errors', {})
+
+          if (options.onSuccess) {
+            return options.onSuccess(page)
+          }
+        },
+      })
+    },
+    post(url, options) {
+      form.submit('post', url, options)
+    },
+    put(url, options) {
+      form.submit('put', url, options)
+    },
+    patch(url, options) {
+      form.submit('patch', url, options)
+    },
+    delete(url, options) {
+      form.submit('delete', url, options)
+    },
+    reset(fields) {
+      const args = Array.isArray(fields)
+        ? fields
+        : Object.values(arguments)
+
+      if (args.length === 0) {
+        form.data = { ... form.defaults }
+      } else {
+        Object.assign(form.data, Object.keys(form.defaults)
+          .filter(key => args.includes(key))
+          .reduce((carry, key) => {
+            carry[key] = form.defaults[key]
+            return carry
+          }, {})
+        )
```

Comment:
I think readability could be improved a bit if both arguments are on a separate line. Otherwise it's a bit confusing what `.filter` and `.reduce` are working on.

```js
Object.assign(
    form.data,
    Object.keys(form.defaults)
        .filter(key => args.includes(key))
        .reduce((carry, key) => {
            carry[key] = form.defaults[key]
            return carry
        }, {})
)
```

Or even:
```js
Object.assign(
    form.data,
    Object
        .keys(form.defaults)
        .filter(key => args.includes(key))
        .reduce((carry, key) => {
            carry[key] = form.defaults[key]
            return carry
        }, {})
)
```